erpnext/stock/doctype/stock_entry/stock_entry.py

```diff
@@ -346,14 +353,14 @@ def validate_purchase_order(self):
 
 	def validate_bom(self):
 		for d in self.get('items'):
-			if d.bom_no:
+			if d.bom_no and (d.t_warehouse != self.pro_doc.scrap_warehouse):
```

Comment:
`getattr(self, "pro_doc", frappe._dict()).scrap_warehouse`

Needed for repacking.